pylint/checkers/classes/special_methods_checker.py

```diff
@@ -16,11 +22,16 @@
     only_required_for_messages,
     safe_infer,
 )
+from pylint.lint.pylinter import PyLinter
 
 NEXT_METHOD = "__next__"
 
 
-def _safe_infer_call_result(node, caller, context=None):
+def _safe_infer_call_result(
+    node: nodes.FunctionDef,
+    caller: nodes.FunctionDef,
+    context: InferenceContext | None = None,
+) -> Iterator[nodes.NodeNG | Uninferable | None] | None:
```

Comment:
See previous comments, why isn't this `InferenceResult`?

Reply:
I did not take most comment into account yet just the one regarding f-string
@Pierre-Sassoulas
[Use an fstring so tuples with more than 2 element do not crash](https://github.com/PyCQA/pylint/pull/7190/commits/1a066943f544e002287fc85c1ab0549350e84fe6)
[1a06694](https://github.com/PyCQA/pylint/pull/7190/commits/1a066943f544e002287fc85c1ab0549350e84fe6)
@Pierre-Sassoulas
[Use standard import for ImportNode](https://github.com/PyCQA/pylint/pull/7190/commits/e69f75e09e2bfafea3fb602d92d63e7c9e76e045)
[e69f75e](https://github.com/PyCQA/pylint/pull/7190/commits/e69f75e09e2bfafea3fb602d92d63e7c9e76e045)